--- mono-2.6.4/mcs/build/config-default.make	2010-04-22 16:38:03.000000000 +0100
+++ mono-2.6.4/mcs/build/config-default-new.make	2010-04-27 00:29:59.000000000 +0100
@@ -19,7 +19,7 @@
 endif
 prefix = /usr/local
 exec_prefix = $(prefix)
-mono_libdir = $(exec_prefix)/lib
+mono_libdir = $(libdir)
 sysconfdir = $(prefix)/etc
 #RUNTIME = mono
 RUNTIME = false
--- mono-2.6.4/configure.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/configure-new.in	2010-04-27 00:32:41.000000000 +0100
@@ -2679,7 +2679,7 @@
     echo "prefix=$prefix" > $srcdir/$mcsdir/build/config.make
     echo "exec_prefix=$exec_prefix" >> $srcdir/$mcsdir/build/config.make
     echo "sysconfdir=$sysconfdir" >> $srcdir/$mcsdir/build/config.make
-    echo 'mono_libdir=${exec_prefix}/lib' >> $srcdir/$mcsdir/build/config.make
+    echo 'mono_libdir=@libdir@' >> $srcdir/$mcsdir/build/config.make
     echo 'MCS_FLAGS = $(PLATFORM_DEBUG_FLAGS)' >> $srcdir/$mcsdir/build/config.make
     echo 'IL_FLAGS = /debug' >> $srcdir/$mcsdir/build/config.make
     echo "RUNTIME = $mono_build_root/runtime/mono-wrapper" >> $srcdir/$mcsdir/build/config.make
@@ -2719,7 +2719,7 @@
     if test x$cross_compiling = xno && test x$enable_olive_build != xno; then
       echo "prefix=$prefix" > $srcdir/$olivedir/build/config.make
       echo "exec_prefix=$exec_prefix" >> $srcdir/$olivedir/build/config.make
-      echo 'mono_libdir=${exec_prefix}/lib' >> $srcdir/$olivedir/build/config.make
+      echo 'mono_libdir=@libdir@' >> $srcdir/$olivedir/build/config.make
       echo 'MCS_FLAGS = $(PLATFORM_DEBUG_FLAGS)' >> $srcdir/$olivedir/build/config.make
       echo "RUNTIME = $mono_build_root/runtime/mono-wrapper" >> $srcdir/$olivedir/build/config.make
       echo "MONO_VERSION = $myver" >> $srcdir/$olivedir/build/config.make
--- mono-2.6.4/scripts/Makefile.am	2010-04-22 16:37:43.000000000 +0100
+++ mono-2.6.4/scripts/Makefile-new.am	2010-04-27 00:42:28.000000000 +0100
@@ -176,7 +176,7 @@
 endif
 else
 plat_bindir = $(bindir)
-mono_instdir = $(prefix)/lib/mono
+mono_instdir = $(libdir)/mono
 endif
 
 
--- mono-2.6.4/data/cecil.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/cecil.pc-new.in	2010-04-27 00:33:58.000000000 +0100
@@ -1,4 +1,4 @@
-assemblies_dir=@prefix@/lib/mono
+assemblies_dir=@libdir@/mono
 Libraries=${assemblies_dir}/gac/Mono.Cecil/0.6.9.0__0738eb9f132ed756/Mono.Cecil.dll
 
 Name: Mono Internal -- Do not use.
--- mono-2.6.4/data/mint.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mint.pc-new.in	2010-04-27 00:34:27.000000000 +0100
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${pcfiledir}/../..
-libdir=${prefix}/lib
+libdir=@libdir@
 includedir=${prefix}/include/mono-@API_VER@
 
 
--- mono-2.6.4/data/mono.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mono.pc-new.in	2010-04-27 00:34:57.000000000 +0100
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${pcfiledir}/../..
-libdir=${prefix}/@reloc_libdir@
+libdir=@libdir@
 includedir=${prefix}/include/mono-@API_VER@
 sysconfdir=@sysconfdir@
 
--- mono-2.6.4/data/mono.web.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mono.web.pc-new.in	2010-04-27 00:40:27.000000000 +0100
@@ -1,4 +1,4 @@
 Name: Mono.Web
 Description: Mono ASP.NET extension classes
 Version: 2.0.0.0
-Libs: -r:@prefix@/lib/mono/gac/Mono.Web/2.0.0.0__0738eb9f132ed756/Mono.Web.dll
+Libs: -r:@libdir@/mono/gac/Mono.Web/2.0.0.0__0738eb9f132ed756/Mono.Web.dll
--- mono-2.6.4/data/mono-cairo.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mono-cairo.pc-new.in	2010-04-27 00:36:06.000000000 +0100
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${pcfiledir}/../..
-libdir=${prefix}/@reloc_libdir@
+libdir=@libdir@
 includedir=${prefix}/include
 
 Name: Mono.Cairo
--- mono-2.6.4/data/monodoc.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/monodoc.pc-new.in	2010-04-27 00:36:45.000000000 +0100
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@prefix@/lib
+libdir=@libdir@
 sourcesdir=${libdir}/monodoc/sources
 
 Name: Monodoc
--- mono-2.6.4/data/mono-lineeditor.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mono-lineeditor.pc-new.in	2010-04-27 00:37:17.000000000 +0100
@@ -1,5 +1,5 @@
 prefix=${pcfiledir}/../..
-assemblies_dir=${prefix}/lib/mono-source-libs
+assemblies_dir=@libdir@/mono-source-libs
 Sources=${assemblies_dir}/getline.cs
 
 Name: Mono.Terminal.LineEditor
--- mono-2.6.4/data/mono-nunit.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mono-nunit.pc-new.in	2010-04-27 00:37:48.000000000 +0100
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 
 
 Name: Mono NUnit
--- mono-2.6.4/data/mono-options.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/mono-options.pc-new.in	2010-04-27 00:38:21.000000000 +0100
@@ -1,5 +1,5 @@
 prefix=${pcfiledir}/../..
-assemblies_dir=${prefix}/lib/mono-source-libs
+assemblies_dir=@libdir@/mono-source-libs
 Sources=${assemblies_dir}/Options.cs
 
 Name: Mono.Options
--- mono-2.6.4/data/system.web.extensions.design_1.0.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/system.web.extensions.design_1.0.pc-new.in	2010-04-27 00:38:43.000000000 +0100
@@ -1,4 +1,4 @@
 Name: System.Web.Extensions.Design
 Description: System.Web.Extensions.Design ASP.NET 2.0 add-on
 Version: 1.0.61025.0
-Libs: -r:@prefix@/lib/mono/gac/System.Web.Extensions.Design/1.0.61025.0__31bf3856ad364e35/System.Web.Extensions.Design.dll
+Libs: -r:@libdir@/mono/gac/System.Web.Extensions.Design/1.0.61025.0__31bf3856ad364e35/System.Web.Extensions.Design.dll
--- mono-2.6.4/data/system.web.extensions_1.0.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/system.web.extensions_1.0.pc-new.in	2010-04-27 00:39:00.000000000 +0100
@@ -1,4 +1,4 @@
 Name: System.Web.Extensions
 Description: System.Web.Extensions ASP.NET 2.0 add-on
 Version: 1.0.61025.0
-Libs: -r:@prefix@/lib/mono/gac/System.Web.Extensions/1.0.61025.0__31bf3856ad364e35/System.Web.Extensions.dll
+Libs: -r:@libdir@/mono/gac/System.Web.Extensions/1.0.61025.0__31bf3856ad364e35/System.Web.Extensions.dll
--- mono-2.6.4/data/system.web.mvc.pc.in	2010-04-22 16:37:47.000000000 +0100
+++ mono-2.6.4/data/system.web.mvc.pc-new.in	2010-04-27 00:39:16.000000000 +0100
@@ -1,4 +1,4 @@
 Name: System.Web.Mvc
 Description: System.Web.Mvc - ASP.NET MVC
 Version: 1.0.0.0
-Libs: -r:@prefix@/lib/mono/gac/System.Web.Mvc/1.0.0.0__31bf3856ad364e35/System.Web.Mvc.dll
+Libs: -r:@libdir@/mono/gac/System.Web.Mvc/1.0.0.0__31bf3856ad364e35/System.Web.Mvc.dll

