--- mono-1.9.1/data/cecil.pc.in	2008-01-29 22:02:46.000000000 +0000
+++ mono-1.9.1/data/cecil.pc-new.in	2008-04-21 15:05:05.000000000 +0100
@@ -1,4 +1,4 @@
-assemblies_dir=@prefix@/lib/mono
+assemblies_dir=@libdir@/mono
 Libraries=${assemblies_dir}/gac/Mono.Cecil/0.6.8.8607__0738eb9f132ed756/Mono.Cecil.dll
 
 Name: Mono Internal -- Do not use.
--- mono-1.9.1/data/mint.pc.in	2007-11-08 22:07:46.000000000 +0000
+++ mono-1.9.1/data/mint.pc-new.in	2008-04-21 15:05:55.000000000 +0100
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${pcfiledir}/../..
-libdir=${prefix}/lib
+libdir=@libdir@
 includedir=${prefix}/include/mono-@API_VER@
 
 
--- mono-1.9.1/data/mono-cairo.pc.in	2007-11-08 22:07:46.000000000 +0000
+++ mono-1.9.1/data/mono-cairo.pc-new.in	2008-04-21 15:06:41.000000000 +0100
@@ -6,4 +6,4 @@
 Name: Mono.Cairo
 Description: Cairo bindings for Mono
 Version: @VERSION@
-Libs: -r:${prefix}/lib/mono/1.0/Mono.Cairo.dll
+Libs: -r:${libdir}/mono/1.0/Mono.Cairo.dll
