Index: mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/DirectoryScanner.cs
===================================================================
--- mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/DirectoryScanner.cs	(revision 158080)
+++ mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/DirectoryScanner.cs	(revision 158081)
@@ -114,10 +114,12 @@
 					if (!excludedItems.ContainsKey (fi.FullName)) {
 						TaskItem item = new TaskItem (include_item);
 						item.ItemSpec = fi.FullName;
-						string rec_dir = Path.GetDirectoryName (fi.FullName.Substring (wildcard_offset));
-						if (rec_dir.Length > 0)
-							rec_dir += Path.DirectorySeparatorChar;
-						item.SetMetadata ("RecursiveDir", rec_dir);
+						if (wildcard_offset >= 0) {
+							string rec_dir = Path.GetDirectoryName (fi.FullName.Substring (wildcard_offset));
+							if (rec_dir.Length > 0)
+								rec_dir += Path.DirectorySeparatorChar;
+							item.SetMetadata ("RecursiveDir", rec_dir);
+						}
 						includedItems.Add (item);
 					}
 				}
