--- mono-1.2.6/mcs/build/config-default.make	2007-11-08 22:08:59.000000000 +0000
+++ mono-1.2.6/mcs/build/config-default-new.make	2007-12-14 22:25:20.000000000 +0000
@@ -17,7 +17,7 @@
 CFLAGS = -g -O2
 prefix = /usr/local
 exec_prefix = $(prefix)
-mono_libdir = $(exec_prefix)/lib
+mono_libdir = $(libdir)
 sysconfdir = $(prefix)/etc
 #RUNTIME = mono
 RUNTIME = false
--- mono-1.2.6/configure.in	2007-11-08 22:07:57.000000000 +0000
+++ mono-1.2.6/configure-new.in	2007-12-14 22:56:17.000000000 +0000
@@ -2182,7 +2182,7 @@
     echo "prefix=$prefix" > $srcdir/$mcsdir/build/config.make
     echo "exec_prefix=$exec_prefix" >> $srcdir/$mcsdir/build/config.make
     echo "sysconfdir=$sysconfdir" >> $srcdir/$mcsdir/build/config.make
-    echo 'mono_libdir=${exec_prefix}/lib' >> $srcdir/$mcsdir/build/config.make
+    echo 'mono_libdir=@libdir@' >> $srcdir/$mcsdir/build/config.make
     echo 'MCS_FLAGS = $(PLATFORM_DEBUG_FLAGS)' >> $srcdir/$mcsdir/build/config.make
     echo 'IL_FLAGS = /debug' >> $srcdir/$mcsdir/build/config.make
     echo "RUNTIME = $mono_build_root/runtime/mono-wrapper" >> $srcdir/$mcsdir/build/config.make
@@ -2215,7 +2215,7 @@
     if test x$cross_compiling = xno && test x$enable_olive_build != xno; then
       echo "prefix=$prefix" > $srcdir/$olivedir/build/config.make
       echo "exec_prefix=$exec_prefix" >> $srcdir/$olivedir/build/config.make
-      echo 'mono_libdir=${exec_prefix}/lib' >> $srcdir/$olivedir/build/config.make
+      echo 'mono_libdir=@libdir@' >> $srcdir/$olivedir/build/config.make
       echo 'MCS_FLAGS = $(PLATFORM_DEBUG_FLAGS)' >> $srcdir/$olivedir/build/config.make
       echo "RUNTIME = $mono_build_root/runtime/mono-wrapper" >> $srcdir/$olivedir/build/config.make
       echo "MONO_VERSION = $myver" >> $srcdir/$olivedir/build/config.make
--- mono-2.2/data/cecil.pc.in	2008-11-18 21:10:32.000000000 +0000
+++ mono-2.2/data/cecil.pc-new.in	2008-11-18 21:12:20.000000000 +0000
@@ -1,4 +1,4 @@
-assemblies_dir=@prefix@/lib/mono
+assemblies_dir=@libdir@/mono
 Libraries=${assemblies_dir}/gac/Mono.Cecil/0.6.9.0__0738eb9f132ed756/Mono.Cecil.dll
 
 Name: Mono Internal -- Do not use.
--- mono-2.2/data/mint.pc.in	2008-11-18 21:10:32.000000000 +0000
+++ mono-2.2/data/mint.pc-new.in	2008-11-18 21:12:55.000000000 +0000
@@ -1,5 +1,5 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${pcfiledir}/../..
-libdir=${prefix}/lib
+libdir=@libdir@
 includedir=${prefix}/include/mono-@API_VER@
 
--- mono-2.2/data/mono-lineeditor.pc.in	2008-11-11 04:56:15.000000000 +0000
+++ mono-2.2/data/mono-lineeditor.pc-new.in	2008-11-18 21:14:18.000000000 +0000
@@ -1,5 +1,5 @@
 prefix=${pcfiledir}/../..
-assemblies_dir=${prefix}/lib/mono-source-libs
+assemblies_dir=@libdir@/mono-source-libs
 Sources=${assemblies_dir}/getline.cs
 
 Name: Mono.Terminal.LineEditor
--- mono-2.2/data/mono-nunit.pc.in	2008-11-11 04:56:15.000000000 +0000
+++ mono-2.2/data/mono-nunit.pc-new.in	2008-11-18 21:14:34.000000000 +0000
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 
 
 Name: Mono NUnit
--- mono-2.2/data/mono-options.pc.in	2008-11-11 04:56:15.000000000 +0000
+++ mono-2.2/data/mono-options.pc-new.in	2008-11-18 21:14:49.000000000 +0000
@@ -1,5 +1,5 @@
 prefix=${pcfiledir}/../..
-assemblies_dir=${prefix}/lib/mono-source-libs
+assemblies_dir=@libdir@/mono-source-libs
 Sources=${assemblies_dir}/Options.cs
 
 Name: Mono.Options
--- mono-2.2/data/monodoc.pc.in	2008-11-13 23:22:27.000000000 +0000
+++ mono-2.2/data/monodoc.pc-new.in	2008-11-26 12:39:50.000000000 +0000
@@ -1,9 +1,9 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@prefix@/lib
+libdir=@libdir@
 sourcesdir=${libdir}/monodoc/sources
 
 Name: Monodoc
 Description: Monodoc - Mono Documentation Tools
 Version: @VERSION@
-Libs: /r:${prefix}/lib/mono/monodoc/monodoc.dll 
+Libs: /r:${libdir}/mono/monodoc/monodoc.dll 
--- mono-2.2/data/mono.pc.in	2008-09-10 17:12:13.000000000 +0100
+++ mono-2.2/data/mono.pc-new.in	2008-11-26 12:47:07.000000000 +0000
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${pcfiledir}/../..
-libdir=${prefix}/@reloc_libdir@
+libdir=@libdir@
 includedir=${prefix}/include/mono-@API_VER@
 sysconfdir=@sysconfdir@

--- mono-2.2/scripts/Makefile.am	2008-11-18 19:19:30.000000000 +0000
+++ mono-2.2/scripts/Makefile-new.am	2008-11-18 21:00:50.000000000 +0000
@@ -174,7 +174,7 @@
 endif
 else
 plat_bindir = $(bindir)
-mono_instdir = $(prefix)/lib/mono
+mono_instdir = $(libdir)/mono
 endif

 
--- mono-2.4/data/mono-cairo.pc.in	2009-02-13 23:34:32.000000000 +0000
+++ mono-2.4/data/mono-cairo.pc-new.in	2009-03-07 23:32:13.000000000 +0000
@@ -1,9 +1,9 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${pcfiledir}/../..
-libdir=${prefix}/@reloc_libdir@
+libdir=@libdir@
 includedir=${prefix}/include
 
 Name: Mono.Cairo
 Description: Cairo bindings for Mono
 Version: @VERSION@
-Libs: -r:${prefix}/lib/mono/1.0/Mono.Cairo.dll
+Libs: -r:${libdir}/mono/1.0/Mono.Cairo.dll
